src/oop/special_class_method/medium/q11/MainTest.java

```diff
@@ -0,0 +1,236 @@
+            _assertEquals(size, testClass.getFieldValue(testInstance, attributeName1), modifiedClassMessage);
+            _assertEquals(colour, testClass.getFieldValue(testInstance, attributeName2), modifiedClassMessage);
+            assertTrue(testClass.hasModifier(classes, "public"), modifiedClassMessage);
+
+            Object getMethodOutput = testClass.callMethod(getAttributeMethodName0, testInstance);
+            modifiedClassMessage = "You have modified the provided portions of class %s method. Please revert them to the original state.";
+            assertTrue(testClass.hasModifier(getAttributeMethodName0, null, "public"),
+                    String.format(modifiedClassMessage, getAttributeMethodName0));
+            _assertEquals(temp, getMethodOutput,
+                    String.format(modifiedClassMessage, getAttributeMethodName0));
+
+            getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+            assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"),
+                    String.format(modifiedClassMessage, getAttributeMethodName1));
+            _assertEquals(size, getMethodOutput,
+                    String.format(modifiedClassMessage, getAttributeMethodName1));
+
+            getMethodOutput = testClass.callMethod(getAttributeMethodName2, testInstance);
+            assertTrue(testClass.hasModifier(getAttributeMethodName2, null, "public"),
+                    String.format(modifiedClassMessage, getAttributeMethodName2));
+            _assertEquals(colour, getMethodOutput,
+                    String.format(modifiedClassMessage, getAttributeMethodName2));
+
+            Object[][] setMethodArguments = {
+                    {updatedValue, double.class}
+            };
+            Class<?>[] methodModifierClasses = {
+                    double.class
+            };
+            assertTrue(testClass.hasMethod(setAttributeMethodName0, methodModifierClasses, Void.TYPE),
+                    String.format(modifiedClassMessage, setAttributeMethodName0));
+            testClass.callMethod(setAttributeMethodName0, setMethodArguments, testInstance);
+            assertTrue(testClass.hasModifier(setAttributeMethodName0, methodModifierClasses, "public"),
+                    String.format(modifiedClassMessage, setAttributeMethodName0));
+
+            _assertEquals(updatedValue, testClass.getFieldValue(testInstance, attributeName0),
+                    String.format(modifiedClassMessage, setAttributeMethodName0));
+        }
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(3.2, 3.4, "Red"),
+                Arguments.of(432.3, 10.2, "Black"),
+                Arguments.of(1023.4, 11.0, "Blue"),
+                Arguments.of(434.3, 43.4, "White"),
+                Arguments.of(43.34, 7, "Yellow"),
+                Arguments.of(100, 7, "Yellow")
```

Comment:
As you have the change limit in the class, you should use that to code these test cases. For example, you could use **changeLimit** + 0.01, **changeLimit**, and **changeLimit** - 0.01.